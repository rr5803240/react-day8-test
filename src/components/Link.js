import React from 'react'

const STATUS = {
    HOVERED:'hovered',
    NORMAL:'normal'
}

export default class Link extends React.Component{
    constructor(props){
        super(props)
        this.state ={
            class: STATUS.NORMAL,
        }

        this.mouseEnter =  this.mouseEnter.bind(this)
        this.mouseLeave = this.mouseLeave.bind(this)
    }

    mouseEnter(){
         this.setState({class: STATUS.HOVERED})
    }

    mouseLeave(){
        this.setState({class: STATUS.HOVERED})
   }
   render(){
       return(
           <a className={this.state.class} href={this.props.page || '#'} onMouseEnter={this.mouseEnter} onMouseLeave={this.mouseLeave}>{this.props.children}</a>
       )
   }
}