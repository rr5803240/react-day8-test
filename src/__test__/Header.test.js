import React from 'react';
import Link from '../components/Link';
import renderer from 'react-test-renderer'



    test('Link changes the clas when hovered',()=>{
        const component = renderer.create( <Link page="http://www.google.com">Google</Link>,)
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        tree.props.onMouseEnter();
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        tree.props.onMouseLeave();
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
