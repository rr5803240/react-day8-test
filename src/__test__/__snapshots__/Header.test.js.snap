// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the clas when hovered 1`] = `
<a
  className="normal"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link changes the clas when hovered 2`] = `
<a
  className="hovered"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link changes the clas when hovered 3`] = `
<a
  className="hovered"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;
